core/src/test/java/com/linecorp/armeria/server/RequestIdHeadersTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestId;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+import com.linecorp.armeria.testing.server.ServiceRequestContextCaptor;
+
+class RequestIdHeadersTest {
+
+    static final String REQUEST_ID_KEY = "X-RequestId";
+    static final String REQUEST_ID_VAL = "123";
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.service("/configured",
+                       (ctx, req) -> HttpResponse.of(200))
+              .requestIdGenerator((ctx) -> RequestId.of(
+                      Long.parseLong(ctx.headers().get(REQUEST_ID_KEY))));
```

Comment:
Could we add a test that ensures a request ID is created even if a custom `requestIdGenerator` raises an exception

Reply:
Thank you for your comment! 
I modified second test to a case where requestIdGenerator throws an exception.